tooling/nargo/src/lib.rs

```diff
@@ -100,6 +100,27 @@ pub fn prepare_package(package: &Package) -> (Context, CrateId) {
     (context, crate_id)
 }
 
+/// Prepares a package from a source string
+/// This is useful for situations when we don't need dependencies
+/// and just need to operate on single file.
+///
+/// Use case for this is the LSP server and code lenses
+/// which operate on single file and need to understand this file
+/// in order to offer code lenses to the user
+pub fn prepare_source(source: String) -> (Context, CrateId) {
+    let root = Path::new("");
+    let mut file_manager = FileManager::new(root);
+    let root_file_id = file_manager.add_file_with_source(Path::new("main.nr"), source).expect(
+        "Adding source buffer to file manager should never fail when file manager is empty",
+    );
+
+    let mut context = Context::new(file_manager);
+
+    let root_crate_id = context.crate_graph.add_crate_root(root_file_id);
+
+    (context, root_crate_id)
+}
+
```

Comment:
Seems like this would be better placed in the LSP as well if that's the only usecase we have for it so far.

Reply:
This is grouping of functionality in the same place, it creates library which can be reused in different contexts.

Reply:
It's unclear whether we're actually going to use this elsewhere. If we do need it then we can easily move it to a library but it's unnecessary now and just breaks encapsulation.

Reply:
isn't that the point of library - where you don't necessarily know when or if you'll use it but you leave it there for convenience? People are moving or worse - duplication code because it's not where it belongs.

Reply:
No, I would disagree with that usage of a library. If we put any code which we might conceivably use again elsewhere at some unspecified point in the future there immediately, then we back ourselves into a corner of having a ton of public functions which we can't change without it being breaking. It also requires us to jump around the monorepo unnecessarily to follow the execution logic.

Reply:
we do that jumping already - nature of the beast, we should not distribute code in too many places, or this common place does not make that much sense

Reply:
1 🔫 2 - I accept :)